cores/esp8266/core_esp8266_main.cpp

```diff
@@ -112,6 +112,11 @@ extern "C" IRAM_ATTR void esp_schedule() {
     ets_post(LOOP_TASK_PRIORITY, 0, 0);
 }
 
+extern "C" IRAM_ATTR void esp_resume() {
+    esp_schedule();
+    esp_yield();
+}
+
```

Comment:
How about commenting the choice of this name (is it interrupting tasks calling (`delay()`) from cont stack) and also that it is equivalent to `delay(0)` ?

Reply:
If new function names are to be added, with restrictions on usage, like "use esp_break() if code is called from SYS", I suggest adding `_from_sys` in its name to make it clear.

Reply:
`yield()` historically panics on purpose when it is called from SYS.
`esp_break()` is exactly `yield()` without `panic()`, callable from both cont and sys.
Considering that  the `yield()`'s `panic()` is avoided by `delay(0)` or its new flavour `esp_break()`, then we may just update `yield()` to not panic no ?

Reply:
I don't know what happened, but you must not have available the current sources somehow.
The actual comment now states, I think I adapted it after an earlier discussion:
```
// Replacement for delay(0). In CONT, same as yield(). Whereas yield() panics
// in SYS, esp_break() is safe to call and only schedules CONT. Use yield()
// whereever only called from CONT, use esp_break() if code is called from SYS
// or both CONT and SYS.
```
Meaning, `esp_break()` is intended for both SYS and CONT.
With regard to `yield()` panicking in SYS, whereas `delay(0)` does not and `esp_break()` of course does neither, I think this was discussed and it was stated that `yield()` shall intentionally continue panicking when called from anywhere else but CONT.

Reply:
I forgot to disable my previous answer before the last one. Sorry that this confused you.

So you propose `esp_break()` for both sys and cont because `yield()` is for cont only, while they both do the same job, like `delay(0)` which currently is used to replace `yield()` everywhere where it is needed for sys and cont.
Then, you replace `delay(0)` by the new sys+cont `yield()` taste.
So we have now `yield()`, `delay(0)`, `esp_break()` ?
Why not simply `yield()` with a comment ?

If the earlier discussion you refer to is [that one](https://github.com/esp8266/Arduino/pull/7146#pullrequestreview-371951203), it also says:
> We could fix it for a major release

@devyte maybe you can elaborate on this:
> The expressibility issue has been discussed before (a long time ago shortly after I arrived). It was deemed valid

Reply:
@d-a-v Thanks, one can never look too many times at the code. I'm going to explain next that it's a bit different than you think, but that wasn't helped by me replacing `delay(0)` by `yield()` on time too often - instead of  by `esp_break()` (https://github.com/esp8266/Arduino/pull/7148/commits/5c39094e4831e6e937df47eae2ebe991b34c2130).

> Then, you replace delay(0) by the new sys+cont yield() taste.
So we have now yield(), delay(0), esp_break() ?
Why not simply yield() with a comment ?

`Delay(0)` is either pointless, in those places where the code runs only ever from CONT, so for final clarity and to let the runtime `panic` to express that contract, I'm replacing it by `yield()`. Again, only in those places.
Wherever code may run from SYS (and perhaps from CONT, too), the new `esp_break()` must be used to replace `delay(0)`. This is to make the intention clear not to `panic` in SYS, which I find obfuscated by a zero time `delay(0)` call.
As you can see, `yield()` (intentional panic in SYS) is not the same as `esp_break()`, and my reservations about `delay(0)` I've explained above.